### PR TITLE
Refactor to use isEqual:

### PR DESCRIPTION
**Pull request by zenangst**

### CODE REVIEW

File: Source/Others/DDMathEvaluator+FORM.m [hunk 1]
```diff
@@ -38,7 +38,9 @@ + (NSDictionary *)hyp_directoryFunctions {
 
             otherValue = (variables[expression.variable]) ?: expression.variable;
 
-            if (![baseValue isEqualToString:otherValue]) {
+            BOOL baseValueIsEqual = ([baseValue isKindOfClass:[NSString class]] &&
+                                     ![baseValue isEqualToString:otherValue]);
```
Comment (3lvis) [inline Source/Others/DDMathEvaluator+FORM.m]: What about numbers? Should we add another `if` in case it's a number?

Comment (zenangst) [inline Source/Others/DDMathEvaluator+FORM.m]: You don't need `equals` to support numbers. You could do `eq($age,15)` or just `$age == 15` (I think).

Comment (zenangst) [inline Source/Others/DDMathEvaluator+FORM.m]: For reference : https://github.com/davedelong/DDMathParser/wiki/Built-in-Functions

Comment (zenangst) [inline Source/Others/DDMathEvaluator+FORM.m]: ping @3lvis @vadymmarkov

Comment (3lvis) [inline Source/Others/DDMathEvaluator+FORM.m]:
```objc
BOOL baseValueIsEqual = (([baseValue isKindOfClass:[NSString class]] &&
                         ![baseValue isEqualToString:otherValue]) || 
                         ([baseValue isKindOfClass:[NSNumber class]] &&
                         ![baseValue isEqualToNumber:otherValue]));
```

Comment (zenangst) [inline Source/Others/DDMathEvaluator+FORM.m]: My question is, should it be supported?

Comment (3lvis) [inline Source/Others/DDMathEvaluator+FORM.m]: I was wondering too :)

**Comment by 3lvis** [comment]
I guess I will just merge.

**Comment by zenangst** [comment]
@3lvis I would like to add some tests with null values

**Comment by zenangst** [comment]
@3lvis I think it should be good now
